x/datapool/handler.go

```diff
@@ -24,6 +24,12 @@ func NewHandler(k keeper.Keeper) sdk.Handler {
 		case *types.MsgCreatePool:
 			res, err := msgServer.CreatePool(sdk.WrapSDKContext(ctx), msg)
 			return sdk.WrapServiceResult(ctx, res, err)
+		case *types.MsgRegisterNFTContract:
```

Comment:
Anyone can create or modify the ContractAddress, so the ContractAddress may change unintentionally.
I think this should be done through governance or upgradeHandler.

Reply:
I totally agree with you 💯 . 
For now, anyone can deploy and migrate contract (no auth required). We have to make a decision for this.
But, it is out of scope of this PR I think. We can handle it in the next PR.

Reply:
Yes, Grate!